source/d3d10/d3d10_impl_swapchain.cpp

```diff
@@ -110,6 +113,20 @@ bool reshade::d3d10::swapchain_impl::on_init()
 }
 void reshade::d3d10::swapchain_impl::on_reset()
 {
+	if (_backbuffer != nullptr)
+	{
+		unsigned int add_references = 0;
+		// Resident Evil 3 releases all references to the back buffer before calling 'IDXGISwapChain::ResizeBuffers', even ones it does not own
+		// Releasing the references ReShade owns would then make the count negative, which consequently breaks DXGI validation, so reset those references here
+		if (_backbuffer.ref_count() == 0)
+			add_references = _backbuffer == _backbuffer_resolved ? 2 : 1;
+		// Add the reference back that was released because of Unreal Engine 4
+		else if (_backbuffer == _backbuffer_resolved)
+			add_references = 1;
+
+		for (unsigned int i = 0; i < add_references; ++i)
+			_backbuffer->AddRef();
+	}																																	 
```

Comment:
It might not be needed. I just thought that the part was missing on releasing the buffers properly. I was unsure if the code was only related to the comment near to it.

Reply:
Removed it for DX10